doc/aspice_swe3_4/example/iceoryx_component/include/example_component/example_derived_class.hpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef IOX_DOC_EXAMPLE_COMPONENT_EXAMPLE_DERIVED_CLASS_HPP
+#define IOX_DOC_EXAMPLE_COMPONENT_EXAMPLE_DERIVED_CLASS_HPP
+
+#include "source/example_base_class.hpp"
+
+
+namespace example
+{
+/// @brief Short description
+/// @code
+///   minimalistic working example which uses all public methods
+/// @endcode
+/// @swcomponent example_component
+void someFreeFunction();
+
+/// @brief Forward declaration of other class
+class SomeOtherClass;
+
+/// @brief Short description
+/// @details Detailed description
+/// @code
+///
+/// @endcode
+/// @note Important note for user/developer
+/// @swcomponent cpp
+template <typename T>
+class ExampleDerivedClass : public ExampleBaseClass<T>
+{
+  public:
+    /// @brief Short description
+    /// @details Detailed description
+    /// @param[in] a Description of input parameter a
+    /// @param[in] b Description of input parameter b
+    ExampleDerivedClass(const uint64_t a, const uint64_t b) noexcept;
+
+    /// @brief Short description
+    void simpleMethod() const noexcept;
+
+    /// @brief Short description
+    /// @param[in] c Description of input parameter c
+    /// @param[out] d Description of output parameter d
+    /// @return Description of return value
+    uint64_t complexMethod(const uint32_t c, const uint32_t* d) noexcept;
+
+    /// @brief A good example method which sets some kind of speed
+    /// @param[in] speed Description of input parameter speed
+    /// @code
+    ///     myClass.goodExampleMethod(200_kmh); // sets it to 200 km/h
+    ///     myClass.goodExampleMethod(40_ms);   // sets it to 40 m/s
+    /// @endcode
+    void goodExampleMethod(const speed_t speed) noexcept;
+
+    /// @brief Short description
+    /// @pre 	Must be called before another method is called
```

Comment:
Good, demonstrates the way of stating pre and postconditions. I think in the long run we also want (Debug) Macros to state them in the code itself. Essentially asserts, maybe wrapped in something fancy such as ENSURE(conditionExpression, "error message") or PRE(conditionExpression). However, not all preconditions are expressable like this, such as: cannot be called concurrently with XYZ.